libs/pdi-scanner/src/rusb_async/error.rs

```diff
@@ -0,0 +1,59 @@
+// File originally copied from https://github.com/a1ien/rusb/commit/4d955e48394e2a5c116fd4e3c9408c88536ed894,
+// with the license:
+//
+//     Copyright (c) 2015 David Cuddeback
+//                   2019 Ilya Averyanov
+//
+//     Permission is hereby granted, free of charge, to any person obtaining
+//     a copy of this software and associated documentation files (the
+//     "Software"), to deal in the Software without restriction, including
+//     without limitation the rights to use, copy, modify, merge, publish,
+//     distribute, sublicense, and/or sell copies of the Software, and to
+//     permit persons to whom the Software is furnished to do so, subject to
+//     the following conditions:
+//
+//     The above copyright notice and this permission notice shall be
+//     included in all copies or substantial portions of the Software.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+//     MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+//     LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+//     OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+//     WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+// Modifications have been made to better support the use case of this project.
```

Comment:
Not totally sure how to review these modded rusb files. Suggestions?

Reply:
Well, we could:
1. Say we want to keep them exactly as-is but still vendor them.
2. Refer to the git commit as a crate dependency to again keep the original versions as-is.
3. Review the diff manually and then just treat them as "our" files, but still vendor them.
4. Fork `rusb` from the point of the commit and make the changes on our fork. This could either be essentially a git subtree or a separate repo.

I'm kinda inclined toward #3, but I don't have a strong preference.

Reply:
I agree option 3 sounds best. I briefly tried to diff the files when I first reviewed this PR but it was a bit complex and I couldn't immediately identify what changes you added (although I was just using `diff`, so there was no colors/syntax highlighting). If you remember and could give an overview, that might help. Or maybe I just need a better diffing tool

Reply:
I'm going to use the `difft` (difftastic) tool to remind myself what the differences are and then comment based on that.